common/persistence/history_manager.go

```diff
@@ -446,7 +446,7 @@ func (m *executionManagerImpl) ReadHistoryBranchByBatch(
 
 	resp := &ReadHistoryBranchByBatchResponse{}
 	var err error
-	_, resp.History, resp.NextPageToken, resp.Size, err = m.readHistoryBranch(true, request)
+	_, resp.History, resp.TransactionIDs, resp.NextPageToken, resp.Size, err = m.readHistoryBranch(true, request)
```

Comment:
This is getting out of hand. I think we need a struct for the results

Reply:
This is internal method used in two places only. I think looking at signature is faster for reading atm. My another concern would be that we create another "ReadHistoryBranch..." structure in code that will add even more ambiguity in naming than we have now.